### DISCUSSION TITLE
Prompt and usePrompt

### DISCUSSION THREAD

**Discussion by zendesk-olgaczarnecka** [remix-run/react-router#9123] [v5 to v6 Migration]
Hello,

I found in the docs that:
> We will absolutely be working on adding this back in to v6 at some point in the near future

Is there any approximate date / timeframes when we can expect Prompt and usePrompt to be included in v6 ?

**Comment by LeonFedotov** [comment]
@mjackson 🥺

**Comment by ryanflorence** [comment]
That time is now. We're revisiting the constraints and hope to have support again with clearly documented scenarios where it falls short due to limitations of the browser APIs available.

**Comment by cbn-falias** [comment]
Just for completeness:
The unstable hooks are available in v6.
```ts
import { unstable_usePrompt, unstable_useBlocker } from 'react-router-dom';
```
